File: sdk/.net/ardvro.sdk.net/component/sqljson/Response.cs
namespace ardvro.sdk.net.component.sqljson;

using ardvro.core.ext.connection;

public class Response
{
    private object _requestData;
    private string _url;
    private IConnection _connection;

    public Response(object requestData, string url, IConnection connection)
    {
        _requestData = requestData;
        _connection = connection;
        _url = url;
    }

    public void Send<T>(Action<T> callback)
    {
        _connection.Submit(_requestData, _url, callback);
    }

}

File: sdk/android/sqljson/Response.java
package ardvro.component.sqljson;

import ardvro.core.connection.IConnection;
import ardvro.core.connection.IResponseListener;

public class Response
{
    private Object _requestData;
    private String _url;
    private IConnection _connection;

    public Response(Object requestData, String url, IConnection connection)
    {
        _requestData = requestData;
        _connection = connection;
        _url = url;
    }

    public void Send(IResponseListener listener)
    {
        _connection.Submit(_requestData, _url, listener);
    }

}
